src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,65 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE;
+
+    private T[] elementData;
+    private int size;
+
+    public MateStack() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     public void push(T value) {
+        if (value == null) {
+            return;
+        }
+
+        resize();
+        elementData[size] = value;
+        size++;
+    }
+
+    private void resize() {
+        if (size == elementData.length) {
+            int newCapacity = elementData.length * 2;
+            if (newCapacity - MAX_ARRAY_SIZE > 0) {
+                newCapacity = MAX_ARRAY_SIZE;
+            }
 
+            T[] newArray = (T[]) new Object[newCapacity];
+            System.arraycopy(elementData, 0, newArray, 0, elementData.length);
+            elementData = newArray;
+        }
     }
 
     public T peek() {
-        return null;
+        if (isEmpty()) {
+            throw new EmptyStackException();
+        }
```

Comment:
This code is repeated in peek and pop, lets create separate method for it, for example like this (then you can also remove the `isEmpty()` method):

```java
private void checkIfStackEmpty() {
        if (size == 0) {
            throw new EmptyStackException();
        }
    }
```